music21/test/test_expressions.py

```diff
@@ -88,16 +90,33 @@ def testExpandTurns(self):
         realized1 = expressions.realizeOrnaments(n1)
         realized2 = expressions.realizeOrnaments(n2)
         self.assertEqual('C5 D5 C5 B-4 C5', ' '.join(n.pitch.nameWithOctave for n in realized1))
-        self.assertEqual('B4 A#4 B4 C#5 B4', ' '.join(n.pitch.nameWithOctave for n in realized2))
+        self.assertEqual('A#4 B4 C#5 B4', ' '.join(n.pitch.nameWithOctave for n in realized2))
+
+        # total length of realized1 is half-note (ql=2.0): quarter note (half the original note)
+        # followed by 4 turn notes that together add up to the other quarter note.
         self.assertEqual(realized1[0].quarterLength, 1.0)
         self.assertEqual(realized1[1].quarterLength, 0.25)
-        self.assertEqual(realized2[0].quarterLength, 0.5)
+        self.assertEqual(realized1[2].quarterLength, 0.25)
+        self.assertEqual(realized1[3].quarterLength, 0.25)
+        self.assertEqual(realized1[4].quarterLength, 0.25)
+
+        # because turn0.quarterLength defaults to 0.25 (16th-note), each of the four turn
+        # notes is ql=0.25 instead of 0.375 (realize doesn't stretch to fill).
+        self.assertEqual(realized2[0].quarterLength, 0.25)
         self.assertEqual(realized2[1].quarterLength, 0.25)
+        self.assertEqual(realized2[2].quarterLength, 0.25)
+        self.assertEqual(realized2[3].quarterLength, 0.25)
 
+        # Here, delay is still a quarter-note (delay=1.0), and each of the four turn notes
+        # is ql=0.125 (because turn0.quarterLength has been set to 0.125, and realize
+        # doesn't stretch to fit).
```

Comment:
ah -- this is the only substantive issue I see and disagree with -- I think it should be an error to have an ornament where the sum of the realized durations doesn't equal the length of the original.  In the case where there's an explicit delay float set, and an explicit quarterLength for the turn, the last note should be held to the length of the remaining unused duration.

This is common where there's a dotted-eighth + 16th pattern with a delayed turn on the eighth, where the turn is usually executed as 3 notes on the 2nd 16th and the last 16th holds the original note:

C[8.] D[16]  =>  C[16] D[32trip] C[32trip] B[32trip] C[16] D[16]

(which would make a great demo).  It's seen all over the music of Weber.

What I don't want happening is that there's a score with lots of ornaments and someone calls "realizeOrnaments" and now there's gaps in the stream.  That should be an error.

Reply:
OK, I think what you mean is that we _should_ stretch to fit, but only stretch the last note of the turn, leaving the first three at the prescribed `turn.quarterLength` duration.  I'll do that, and take a shot at putting that Weber turn in the doctest.

Reply:
Done.